src/routes/project/components/TextEditor.js

```diff
@@ -0,0 +1,194 @@
+import React, { Component } from 'react';
+import Editor from 'draft-js-plugins-editor';
+import { EditorState, RichUtils, getVisibleSelectionRect, convertToRaw, convertFromRaw } from 'draft-js'
+import createCounterPlugin from 'draft-js-counter-plugin';
+
+const counterPlugin = createCounterPlugin();
+const { CharCounter, WordCounter, CustomCounter } = counterPlugin;
+const plugins = [counterPlugin];
+
+
+class StyleButton extends Component {
+  constructor() {
+    super();
+    this.onToggle = (e) => {
+      e.preventDefault();
+      this.props.onToggle(this.props.style);
+    };
+  }
+
+  render() {
+    let className = 'toolbarItem';
+    if (this.props.active) {
+      className += ' active';
+    }
+    return (
+      <span className={className} onMouseDown={this.onToggle}>
```

Comment:
changed to onClick, also, gonna keep as spans for now given that flexing buttons is a pain and hacky for what I understand (having to nest spans within a span to allow cross-browser flex, unless Im missing something)